src/js/jsx/search/SearchBar.jsx

```diff
@@ -416,15 +429,19 @@ define(function (require, exports, module) {
          * Perform action based on ID
          *
          * @private
-         * @type {Datalist~onChange}
+         * @type {Datalist~onInputChange}
+         * @param {string} id of the selected option
+         * @param {string} searchTerm user input
+         * @return {{dontCloseDialog: boolean=}};
          */
-        _handleChange: function (id) {
-            if (id === null) {
+        _handleChange: function (id, searchTerm) {
+            if (id === null || !id) {
```

Comment:
This is redundant, since if `id === null` then `!id` would suffice. Also the param type annotation says that string is required, so that would imply that it can't be null. Do you know if `id` is actually ever null?